src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,75 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    static final int MAX_NUM_FIGURES = 5;
+    static final int MAX_NUM_FOR_AREA_X = 20;
+    static final int MAX_NUM_FOR_AREA_Y = 20;
+    static final int MAX_NUM_FOR_AREA_Z = 20;
+
+    public Figure getRandomFigure() {
+        Random randomNumber = new Random();
+        int figureNumber = randomNumber.nextInt(MAX_NUM_FIGURES) + 1;
+        if (figureNumber == 1) {
+            return getNewCircle();
+        } else if (figureNumber == 2) {
+            return getNewIsoscelesTrapezoid();
+        } else if (figureNumber == 3) {
+            return getNewRectangle();
+        } else if (figureNumber == 4) {
+            return getNewRightTriangle();
+        } else {
+            return getNewSquare();
+        }
+    }
+
+    private Figure getNewCircle() {
+        ColorSupplier colorSupplier = new ColorSupplier();
+        Random randomNumber = new Random();
+        double numForAreaX = randomNumber.nextInt(MAX_NUM_FOR_AREA_X) + 1;
```

Comment:
```suggestion
        double radius = randomNumber.nextInt(MAX_PROPERTY_VALUE) + 1;
```
same for other figures - use more informative variable names